apps/site/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page.tsx

```diff
@@ -65,15 +65,18 @@ const EntityTypePage: NextPage = () => {
       return;
     }
 
-    const schema = getSchemaFromFormData(data);
+    const newLinksAndProperties = getSchemaFromFormData(data);
+
+    const existingSchema = entityType.schema;
 
     const { data: responseData, error: responseError } =
       await apiClient.updateEntityType({
         versionedUri: entityType.schema.$id,
         schema: {
-          ...schema,
-          properties: schema.properties ?? {},
-          title: schema.title ?? entityType.schema.title,
+          ...existingSchema,
+          links: newLinksAndProperties.links ?? existingSchema.links ?? {},
+          properties:
+            newLinksAndProperties.properties ?? existingSchema.properties ?? {},
```

Comment:
I think ideally we'd just make sure the types of both the `existingSchema` and `newLinksAndProperties` are correct

Reply:
> If we're doing it for these do we want to do it for the other optional fields

I don't think its necessary as we can leave it to the API to make sure we don't get inappropriately `null` fields in the types (since it's an API/db issue that they might). These fallbacks won't themselves be necessary once the `newLinksAndProperties` is properly typed, but at the moment it doesn't guarantee `links` or `properties` is present